src/Extension/EscaperExtension.php

```diff
@@ -132,7 +132,7 @@ public function setEscaper($strategy, callable $callable)
 
         $this->escapers[$strategy] = $callable;
         $callable = function ($string, $charset) use ($callable) {
-            return $callable($this->environment, $string);
+            return $callable($this->environment, $string, $this->environment->getCharset());
```

Comment:
It was indeed a mistake. I don't think we should restore this odd behavior as the callables never got the charset before (so without changing the escapers, the charset is not useful). And as adding the charset does not make migrating easier, I don't think we should support it.

Reply:
Not sure what you mean? The callables did get the charset before.

Reply:
Sorry, I'm stupid sometimes. See #4091 for the fix. Here, we need the charset passed to the escape filter, not the default charset.